luaui/Widgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,277 @@
+local WIDGET_NAME = "Construction Turrets Range Check"
+local WIDGET_VERSION = "1.5"
+--[[
+### VERSIONS ###
+1.0 - initial release, basic
+1.1 - added more command types (reclaim, attack)
+1.2 - added support for queued commands to be flag_processed
+1.3 - fixed a range deviation caused by the game adding model radius
+1.4a - optimization, added LRU cache to reduce the number of calls to the engine
+1.4b - optimization, changed the flag_listening method to await a command instead of polling x'th frame
+1.4c - optimization, added a command limit to prevent the engine from ignoring commands
+1.4d - optimization, replaced loop with GiveOrderToUnitArray, renaiming and adding comments
+1.4e - changed distance calculation from 3D to 2D
+1.4f - removed a Spring.Echo() that was left from debugging.
+1.4g - refactored the code, added more comments, better readability.
+1.5 - fixed edge case that allows to skip the script the execution of the script by switching selection.
+1.5b - main branch integration. Removal of hello Echo.
+]]--
+
+function widget:GetInfo()
+    return {
+        name = WIDGET_NAME,
+        desc = "Stops construction turrets from being assigned to constructions out of reach.",
+        author = "Nehroz",
+        date = "2024.11.01", -- update date.
+        license = "GPL v3",
```

Comment:
we use gpl v2 as the license for widgets. see https://github.com/beyond-all-reason/Beyond-All-Reason/blob/b736c47b1a7d45cfcd205be92c835de734dd4704/luaui/Widgets/gui_gridmenu.lua#L17 as an example